dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -323,10 +323,51 @@ else if (fd.getDataLen() == 2)
                 {
                     this.buffer.putShort(value.shortValue());
                 }
+                else if (fd.getDataLen() == 3)
+                {
+                    this.buffer.put((byte) (value & 0xFF));
+                    this.buffer.put((byte) ((value >> 8) & 0xFF));
+
+                    long signBit = value < 0 ? 0x80L : 0;
+                    this.buffer.put((byte) (((value >> 16) & 0xFF) | signBit));
+                }
                 else if (fd.getDataLen() == 4)
                 {
                     this.buffer.putInt(value.intValue());
                 }
+                else if (fd.getDataLen() == 5)
+                {
+                    this.buffer.put((byte) (value & 0xFF));
+                    this.buffer.put((byte) ((value >> 8) & 0xFF));
+                    this.buffer.put((byte) ((value >> 16) & 0xFF));
+                    this.buffer.put((byte) ((value >> 24) & 0xFF));
+                    
+                    long signBit = value < 0 ? 0x80L : 0;
+                    this.buffer.put((byte) (((value >> 32) & 0xFF) | signBit));
+                }
+                else if (fd.getDataLen() == 6)
+                {
+                    this.buffer.put((byte) (value & 0xFF));
+                    this.buffer.put((byte) ((value >> 8) & 0xFF));
+                    this.buffer.put((byte) ((value >> 16) & 0xFF));
+                    this.buffer.put((byte) ((value >> 24) & 0xFF));
+                    this.buffer.put((byte) ((value >> 32) & 0xFF));
+                    
+                    long signBit = value < 0 ? 0x80L : 0;
+                    this.buffer.put((byte) (((value >> 40) & 0xFF) | signBit));
+                }
+                else if (fd.getDataLen() == 7)
+                {
+                    this.buffer.put((byte) (value & 0xFF));
+                    this.buffer.put((byte) ((value >> 8) & 0xFF));
+                    this.buffer.put((byte) ((value >> 16) & 0xFF));
+                    this.buffer.put((byte) ((value >> 24) & 0xFF));
+                    this.buffer.put((byte) ((value >> 32) & 0xFF));
+                    this.buffer.put((byte) ((value >> 40) & 0xFF));
```

Comment:
Everything looks good, I had to find something:
consider generalizing this logic and exposing it in a function we could call for fd.getDataLen() == 3,5,6,7.
The function could be something like:

    public void moveNBytesToBuffer(value, buffer, nBytes)
    {
        for (int i = 0; i <nBytes; i+=8)
        {
        	this.buffer.put((byte) ((value >> i) & 0xFF));
		}
        long signBit = value < 0 ? 0x80L : 0;
        this.buffer.put((byte) (((value >> nBytes*8) & 0xFF) | signBit));
    }